internal/api/shared/crypto/jwt.go

```diff
@@ -28,6 +38,11 @@ type JWTCustomClaims struct {
 
 type JWT interface {
 	SignedToken(login string) (string, error)
+	// CreateJWTCookies will create two different cookies that contain a piece of the token.
+	// As a reminder, a JWT token has the following structure: header.payload.signature
+	// The first cookie will contain the struct header.payload that can then be manipulated by Javascript
```

Comment:
In which use case do we need the client to manipulate the token payload? If the payload is changed then the token will be tampered and the signature won't match.

Reply:
maybe the word `manipulate` is not the correct one. It's more on the client side, we want to decode the token to know if it is expired (at least) and then later to know which permission have the user so we can modify the UI to reflect these permission.

Reply:
In that case the complete token can be used, as decoding does not require signature validation.

Reply:
yeah I know :). That's what I'm doing. I just store the token in two different cookies based on what I read here: https://stackoverflow.com/a/61106253. And I'm quite convinced to follow.